sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/SelectQuery.cs

```diff
@@ -111,8 +111,13 @@ public override AdtQueryBuilder Build()
         /// <inheritdoc/>
         public override string GetQueryText()
         {
+            // handle passing in null
+            if (_clause == null || _clause.ClauseArgs == null)
```

Comment:
[Null conditional operator](https://docs.microsoft.com/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-) can help you reduce the number of things to check.

```suggestion
        if (_clause?.ClauseArgs == null)
```